### PR TITLE
Various doc fixes

### PR DESCRIPTION
**Pull request by einarf**
A collection doc and project tweaks

* README.md RTD badge
* Add `requirements.txt` for tests (makes instructions simpler)
* Typos and outdated commands / names
* Rewrite building docs
* Basic doc building instructions in `README.md`
* Basic tests instructions in `README.md`
* Additional info in `internal/doc`
* Mission optional test dependencies
* Simplify dev setup / virtualenv instructions

### CODE REVIEW

**Comment by einarf** [comment]
Done for now. Please show no mercy in the review 😄

File: doc/internal/contributing.rst [hunk 1]
```diff
@@ -31,8 +31,12 @@ get the latest version of the code using:
 
 .. code-block:: bash
 
+    # clone over https
     git clone https://github.com/pyglet/pyglet.git
```
Comment (benmoran56) [inline doc/internal/contributing.rst]: Not a bad addition.

File: doc/internal/testing.rst [hunk 2]
```diff
@@ -45,26 +45,26 @@ There are currently 3 types of interactive test cases:
 Running tests
 -------------
 
-The pyglet test suite is based on the `py.test framework <http://pytest.org>`_.
```
Comment (benmoran56) [inline doc/internal/testing.rst]: Good change. It seems like the "dot" has been deprecated since pytest 3.0.

File: tests/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+pytest<6
```
Comment (benmoran56) [inline tests/requirements.txt]: I'm wondering why we would specifically mark the version as <6?  Is there an upcoming change that is known to break, or is it in the effort to not break unexpectedly?

File: doc/internal/doc.rst [hunk 4]
```diff
@@ -118,14 +118,32 @@ Make sure you prepare your environment as stated in :doc:`virtualenv`.
...
+    python setup.py build_sphinx
+
+    # make (make.bat for windows)
+    cd doc
+    make html
+
 HTML Theme
 ----------
 
+.. Note:: The custom theme was disabled in 2019 and replaced with
```
[the remark this answers is not in the document]
Comment (einarf) [reply doc/internal/doc.rst]: Deal. I also think fixing it is very unlikely.

File: tests/requirements.txt (hunk 3 above)
Comment (einarf) [reply tests/requirements.txt]: I normally cap the version like this, but considering this is dev requirements and not even applied consistently I guess we can drop it.

File: doc/internal/testing.rst (hunk 2 above)
Comment (einarf) [reply doc/internal/testing.rst]: Yep. It was also originally scheduled for removal in 6.x as far as I remember.

**Comment by einarf** [comment]
Updated with the changes requested.

**Pull request merged**